bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tf1.py

```diff
@@ -0,0 +1,140 @@
+# Author: Aaron Jomy CERN 08/2024
+
+################################################################################
+# Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.                      #
+# All rights reserved.                                                         #
+#                                                                              #
+# For the licensing terms see $ROOTSYS/LICENSE.                                #
+# For the list of contributors see $ROOTSYS/README/CREDITS.                    #
+################################################################################
+
+r"""
+/**
+\class TF1
+\brief \parblock \endparblock
+\htmlonly
+<div class="pyrootbox">
+\endhtmlonly
+## PyROOT
+
+The TF1 class has several additions for its use from Python, which are also
+available in its subclasses TF2 and TF3.
+
+First, TF1 instance can be inititialised with user-defined Python functions. Given a functor object or a pycallable,
+the following can performed:
+
+\code{.py}
+def func(x, pars):
+    return pars[0] * x[0] * x[0] + x[1] * pars[0]
+
+my_func = ROOT.TF1("my_func", func, -10, 10, npar=2, ndim=2)
+\endcode
+
+Second, after performing the initialisation with a Python functor, the TF1 instance can be evaluated using the Pythonized
+`TF1::EvalPar` function. The pythonization allows passing in the expected data(x {1, 2, .. ndims}) or a dataset (x { {1, 2, .. ndims}, {1, 2, .. ndims}...})
```

Comment:
I understand the distinction between data and dataset, but I believe these terms are a bit too overloaded. Here we are specifically talking about 1D and 2D (numpy) arrays, so I would use the more specific terms instead.